healvis/beam_model.py

```diff
@@ -33,6 +39,51 @@ def airy_disk(za_array, freqs, diameter=15.0, **kwargs):
     return beam
 
 
+def smooth_beam(freqs, beam_array, freq_ls=2.0, noise=1e-10, output_freqs=None):
+    """
+    Smooth a beam across frequency using a Gaussian Process
+
+    Args:
+        freqs : 1D ndarray
+            Frequency array [Hz]
+        beam_array : 2D ndarray
+            Beam map with shape (Nfreqs, Npixels)
+        freq_ls : float
+            Frequency lengthscale to smooth at [MHz]
+        noise : float
+            Noise level. Ideally a small but nonzero value.
+        output_freqs : 1D ndarray
+            Prediction frequencies [Hz]. Default is training frequencies.
+
+    Returns:
+        smooth_beam : 2D ndarray
+            Smoothed beam
+    """
+    assert sklearn_import, "Couldn't import sklearn package. This is required to use beam smoothing functionality."
+
+    # setup kernel
+    kernel = 1**2 * gp.kernels.RBF(freq_ls) + gp.kernels.WhiteKernel(noise)
```

Comment:
1**2 == 1?

Reply:
its part of the gaussian process kernel, it interprets that as its own object, and the sklearn docs do it that way so i figured we should too, but not a big deal in my mind...